src/GovUk.Education.ExploreEducationStatistics.Admin/Requests/BoundaryLevelCreateRequest.cs

```diff
@@ -0,0 +1,50 @@
+#nullable enable
+using FluentValidation;
+using GovUk.Education.ExploreEducationStatistics.Admin.Services.Interfaces;
+using GovUk.Education.ExploreEducationStatistics.Common.Model.Data;
+using Microsoft.AspNetCore.Http;
+using System;
+using System.Linq;
+
+namespace GovUk.Education.ExploreEducationStatistics.Admin.Requests;
+
+public record BoundaryLevelCreateRequest
+{
+    public required GeographicLevel Level { get; init; }
+
+    public required string Label { get; init; }
+
+    public required IFormFile File { get; init; }
+
+    public required DateTime Published { get; init; }
+
+    public class Validator() : AbstractValidator<BoundaryLevelCreateRequest>
+    {
+        public Validator(IBoundaryLevelService boundaryLevelService) : this()
+        {
+            RuleFor(request => request.Level)
+                .NotEmpty();
+
+            RuleFor(request => request.Label)
+                .NotEmpty()
+                .MustAsync(async (label, cancellation) =>
+                {
+                    var existingLevels = await boundaryLevelService.ListBoundaryLevels();
+
+                    return !existingLevels.IsLeft && !existingLevels.Right
+                        .Select(el => el.Label)
+                        .ToList()
+                        .Contains(label);
+                })
```

Comment:
I don't think we've got any examples of using services within validators to know how best to handle a service failure like this off the top of my head, but you could at least simplify this as follows:

```csharp
                .MustAsync(async (label, cancellationToken) =>
                {
                    var existingLevels = await boundaryLevelService.ListBoundaryLevels(cancellationToken);
                    return existingLevels.IsRight && !existingLevels.Right.Any(bl => bl.Label == label);
                })
```